### PR TITLE
Misc cleanup

### PR DESCRIPTION
**Pull request by fingolfin**
- **Fix typo**
- **Move some ZZMatrix code to fmpz_mat.jl**
- **Enable 'f(x)' support where f is an arb/acb/real/complex poly**
- **Remove redundant is_zero_row methods**

For further details please refer to the respective commit messages.

### CODE REVIEW

File: src/flint/fmpz_mat.jl [hunk 1]
```diff
@@ -618,6 +623,58 @@ Reduce the entries of $x$ modulo $y$ and return the result.
...
+  end
+  return nothing
+end
+
+@doc raw"""
+    mod(M::ZZMatrix, p::ZZRingElem) -> ZZMatrix
+
+Reduces every entry modulo $p$, i.e. applies the mod function to every entry.
+"""
+function mod(M::ZZMatrix, p::ZZRingElem)
```
Comment (fingolfin) [inline src/flint/fmpz_mat.jl:649]: I note that there is an existing function `reduce_mod` which seems to do the same, albeit with a direct call to function `fmpz_mat_scalar_mod_fmpz` which I am guessing be more efficient?

What should we do, keep both `reduce_mod` and `mod`? Or merge them? Thoughts, @fieker @thofma ?

**Review by lgoettgens — APPROVED** [review]
apart from the one unresolved question, this looks good

**Comment by lgoettgens** [comment]
Unfortunately we have conflicts with https://github.com/Nemocas/Nemo.jl/pull/1768

**Comment by fingolfin** [comment]
rebased. And I've filed #1773 for `reduce_mod` vs. `mod`, we don't need to settle it in this PR -- at least now the code for both is next to each other, making the duplication a bit more "obvious"

**Comment by lgoettgens** [comment]
> rebased. And I've filed #1773 for `reduce_mod` vs. `mod`, we don't need to settle it in this PR -- at least now the code for both is next to each other, making the duplication a bit more "obvious"

Thanks for both!

**Pull request merged**